### PR TITLE
feat(financial-statements-inao): Add application-api-template module and client

### PR DESCRIPTION
**Pull request by fjandakornid**
# Add application api template module for Financial Statements IANO

This is a redo of #7823 that I couldn't get working after the framework update.

## What

- Added application api template module for Financial Statements INAO
- Moved client from api-module to it's own lib
- Changed connection for Dataverse from sandbox to RE environment
- Removed server-feature-flag

## Why

Needed to get current user's client type from Dataverse in applications x-state onEntry function.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by eirikurn — APPROVED** [review]
LGTM!

File: libs/application/template-api-modules/src/lib/modules/templates/financial-statements-inao/financial-statements-inao.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export class FinancialStatementsInaoTemplateService {
+  constructor(
+    private financialStatementsService: FinancialStatementsInaoClientService,
+  ) {}
+
+  async getUserType({ auth }: TemplateApiModuleActionProps) {
+    const { nationalId } = auth
+
+    if (kennitala.isPerson(nationalId)) {
+      return this.financialStatementsService.getClientType('Einstaklingur')
```
Comment (FridrikMProgramm) [inline libs/application/template-api-modules/src/lib/modules/templates/financial-statements-inao/financial-statements-inao.service.ts:17]: Put 'Einstaklingur' in a constant, shareable with front end?

File: libs/clients/financial-statements-inao/src/lib/financialStatementsInao.config.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { defineConfig } from '@island.is/nest/config'
 
-export const dataverseClientConfig = defineConfig({
+export const FinancialStatementsInaoClientConfig = defineConfig({
   name: 'DataverseClient',
```
Comment (FridrikMProgramm) [inline libs/clients/financial-statements-inao/src/lib/financialStatementsInao.config.ts:4]: Update this name?

**Review by FridrikMProgramm — APPROVED** [review]
Approved with minor musings.

File: libs/application/template-api-modules/src/lib/modules/templates/financial-statements-inao/financial-statements-inao.service.ts (hunk 1 above)
Comment (fjandakornid) [reply libs/application/template-api-modules/src/lib/modules/templates/financial-statements-inao/financial-statements-inao.service.ts:17]: Will probably add this to environment variable in later branch

File: libs/clients/financial-statements-inao/src/lib/financialStatementsInao.config.ts (hunk 2 above)
Comment (fjandakornid) [reply libs/clients/financial-statements-inao/src/lib/financialStatementsInao.config.ts:4]: Will take a look at it in next branch

**Review by obmagnusson — APPROVED** [review]
LGTM

**Review by pshomov — APPROVED** [review]
LGTM

**Pull request merged**